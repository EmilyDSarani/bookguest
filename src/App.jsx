import React from 'react'
import Home from './views/Home/Home'
import Layout from './views/Layout/Layout'
import PrivateRoute from './components/PrivateRoute/PrivateRoute'
import Auth from './components/Auth/Auth'
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
  Redirect
} from "react-router-dom"

export default function App() {
  return (
    <Router>
    <Layout>
      <Switch>
        <Route path="/login">
          <Auth />
        </Route>
        <PrivateRoute path="/">  
          <Home />
        </PrivateRoute>
      </Switch>
    </Layout>
    </Router>
  )
}
