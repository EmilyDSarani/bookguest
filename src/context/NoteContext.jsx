import { useContext, createContext, useState } from 'react'

//Create Context
//Remember that this is the start of our recipe
const NoteContext = createContext()

//Create Provider
//Remember that the provider is giving us access to the recipe book
const NoteProvider = ({ children }) =>{
    const [notes, setNotes] = useState([])
    return <NoteContext.Provider value={{notes, setNotes}}>{children}</NoteContext.Provider>   
}

//Create Custom Hook
const useNotes = () => {
    const context = useContext(NoteContext)

    if(context === undefined){
        throw new Error('useUser needs to be defined in the NoteContext Provider')
    }

    return context
}

export {NoteProvider, useNotes}