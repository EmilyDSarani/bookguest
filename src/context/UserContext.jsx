import { useContext, createContext, useState, useEffect } from 'react'

//Create Context
//Remember that this is the start of our recipe
const UserContext = createContext()

//Create Provider
//Remember that the provider is giving us access to the recipe book
const UserProvider = ({ children }) =>{
    const [user, setUser] = useState(
        //might need to drop a useEffect here later on


    {})
    return <UserContext.Provider value={{user, setUser}}>{children}</UserContext.Provider>   
}

//Create Custom Hook
const useUser = () => {
    const context = useContext(UserContext)

    if(context === undefined){
        throw new Error('useUser needs to be defined in the UserContext Provider')
    }

    return context
}

export {UserProvider, useUser}