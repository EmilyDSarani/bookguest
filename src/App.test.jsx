import { render } from '@testing-library/react';
import App from './App';
import { UserProvider }  from './context/UserContext'
import { NoteProvider } from './context/NoteContext'

test('renders App', () => {
  const {container} = 
  render(
<UserProvider> 
<NoteProvider>      
<App />
</NoteProvider>
</UserProvider>,);
  expect(container).toMatchSnapshot();
});
