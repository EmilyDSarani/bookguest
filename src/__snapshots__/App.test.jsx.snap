// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<div>
  <div>
    <header>
      <p>
        Signed in as 
        <div />
      </p>
    </header>
    <div>
      <h1>
        Thanks !
      </h1>
      <form>
        <div>
          <div>
            <label
              class="labelText"
              for="inputNote"
            >
              Enter Message
            </label>
          </div>
          <div>
            <textarea
              id="inputNote"
              placeholder="Your Note!"
            />
          </div>
        </div>
        <div>
          <button
            class="button"
            type="submit"
          >
            Sign
          </button>
          <button
            type="button"
          >
            Not 
            
            ?
          </button>
        </div>
      </form>
      <div>
        <h1>
          Message Board
        </h1>
        <ul />
      </div>
    </div>
    <footer>
      <p>
        This footer is brought to you by the letter 'F'
      </p>
    </footer>
  </div>
</div>
`;
